docs/en/stack/data-frames/checkpoints.asciidoc

```diff
@@ -0,0 +1,60 @@
+[role="xpack"]
+[[ml-transform-checkpoints]]
+=== {dataframe-transform-cap} checkpoints
+++++
+<titleabbrev>Checkpoints</titleabbrev>
+++++
+
+beta[]
+
+Each time a {dataframe-transform} examines the source indices and creates or
+updates the destination index, it generates a _checkpoint_.
+
+If your {dataframe-transform} runs only once, there is logically only one
+checkpoint. If your {dataframe-transform} runs continuously, however, it creates
+checkpoints as it ingests and transforms new source data.
+
+For example, a {cdataframe-transform}:
+
+. Checks for changes to source indices.
++
+Using a simple periodic timer, the {dataframe-transform} checks for changes to
+the source indices based on sequence IDs. This check is done based on the
+interval defined in the transform's `frequency` property.
++
+If the source indices remain unchanged or if a checkpoint is already in progress
+then it waits for the next timer.
+
+. Identifies which entities have changed.
++
+The {dataframe-transform} searches to see which entities have changed since the
+last time it checked. The transform's `sync` configuration object identifies a
+time field in the source indices. The transform uses the values in that field to
+synchronize the source and destination indices.
+ 
+. Updates the destination index (the {dataframe}) with the changed entities.
++
+--
+The {dataframe-transform} applies changes related to either new or changed
+entities to the destination index.
+//TBD: This seems to imply it doesn't handle deletions. Do we want to clarify that?
+The set of changed entities is paginated. For each page, the
+{dataframe-transform} performs a composite aggregation using a `terms` query.
+//TBD: It's not clear why the previous two sentences are noteworthy from a user point of view.
```

Comment:
The fact that we are using composite aggregations and terms queries show that transforms are built on top of existing elasticsearch constructs. Performance and scalability considerations for aggs and terms queries also apply to data frame transforms, therefore worth being transparent. This is the point of a "how it works" section.

Pagination is relevant to both searching and indexing. This step does not show that paging is relevant to both. Perhaps change last sentence from `After all the changes have been applied, the checkpoint is complete.` to `After all the pages of changes have been applied, the checkpoint is complete.`